File: src/main/java/com/service/PetFoodService.java
package com.service;

import com.model.PetFood;
import com.dao.PetFoodDAO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class PetFoodService {

    @Autowired
    private PetFoodDAO petFoodDAO;

    public List<PetFood> getAllPetFoods() {
        return petFoodDAO.findAll();
    }

    public PetFood getPetFoodById(long foodId) {
        Optional<PetFood> petFood = petFoodDAO.findById(foodId);
        return petFood.orElse(null);
    }

    public List<PetFood> searchPetFoodsByName(String name) {
        return petFoodDAO.findByName(name);
    }

    public List<PetFood> getPetFoodsByType(String type) {
        return petFoodDAO.findByType(type);
    }

    public List<PetFood> getPetFoodsByBrand(String brandName) {
        return petFoodDAO.findByBrand(brandName);
    }

    public PetFood addPetFood(PetFood petFood) {
        return petFoodDAO.save(petFood);
    }

    public PetFood updatePetFood(long foodId, PetFood petFood) {
        if (petFoodDAO.existsById(foodId)) {
            petFood.setFoodId(foodId);
            return petFoodDAO.save(petFood);
        }
        return null;
    }
    

    public PetFood updatePetFoodQuantity(Long food_id, int quantity) {
        PetFood existingPetFood = petFoodDAO.findById(food_id).orElse(null);
        if (existingPetFood != null) {
            existingPetFood.setQuantity(quantity);
            return petFoodDAO.save(existingPetFood);
        }
        return null;
    }
}

File: Admin/services/petfood.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PetFood } from '../../models/petfood';

@Injectable({
  providedIn: 'root'
})
export class PetFoodService {

  private baseUrl = 'http://localhost:8080/api/v1/pet_foods';

  constructor(private http: HttpClient) { }

  getAllPetFoods(): Observable<PetFood[]> {
    return this.http.get<PetFood[]>(`${this.baseUrl}`);
  }

  getPetFoodById(foodId: number): Observable<PetFood> {
    return this.http.get<PetFood>(`${this.baseUrl}/${foodId}`);
  }

  searchPetFoodsByName(name: string): Observable<PetFood[]> {
    return this.http.get<PetFood[]>(`${this.baseUrl}/search?name=${name}`);
  }

  getPetFoodsByType(type: string): Observable<PetFood[]> {
    return this.http.get<PetFood[]>(`${this.baseUrl}/food_type/${type}`);
  }

  getPetFoodsByBrand(brandName: string): Observable<PetFood[]> {
    return this.http.get<PetFood[]>(`${this.baseUrl}/brand/${brandName}`);
  }

  addPetFood(petFood: PetFood): Observable<PetFood> {
    return this.http.post<PetFood>(`${this.baseUrl}/add`, petFood);
  }

  updatePetFood(foodId: number, petFood: PetFood): Observable<PetFood> {
    return this.http.put<PetFood>(`${this.baseUrl}/${foodId}`, petFood);
  }

  updatePetFoodQuantity(foodId: number, quantity: number): Observable<PetFood> {
    return this.http.patch<PetFood>(`${this.baseUrl}/${foodId}/quantity`, { quantity });
  }
}